libraries/app/database_api.cpp

```diff
@@ -624,6 +625,54 @@ vector<optional<account_object>> database_api_impl::get_accounts(const vector<ac
    return result;
 }
 
+vector<limit_order_object> database_api::get_account_limit_orders( const string& name_or_id, const string& base, const string& quote, fc::time_point_sec expire, uint32_t limit)
+{
+   return my->get_account_limit_orders( name_or_id, base, quote, expire, limit );
+}
+
+vector<limit_order_object> database_api_impl::get_account_limit_orders( const string& name_or_id, const string& base, const string& quote, fc::time_point_sec expire, uint32_t limit)
+{
+   vector<limit_order_object>   results;
+   const account_object*        account = nullptr;
+   uint32_t                     count = 0;
+
+   if (std::isdigit(name_or_id[0]))
+      account = _db.find(fc::variant(name_or_id, 1).as<account_id_type>(1));
+   else
+   {
+       const auto& idx = _db.get_index_type<account_index>().indices().get<by_name>();
+       auto itr = idx.find(name_or_id);
+       if (itr != idx.end())
+           account = &*itr;
+   }
+   if (account == nullptr)
+       return results;
+
+   auto assets = lookup_asset_symbols( {base, quote} );
+   auto base_id = assets[0]->id;
+   auto quote_id = assets[1]->id;
+
+   // Add the account's orders
+   auto order_range = _db.get_index_type<limit_order_index>().indices().get<by_account>().equal_range(account->id);
```

Comment:
Perhaps best to add an index for this API? Iterating with `continue` below is inefficient.